frontend/src/users/components/RegisterForm.tsx

```diff
@@ -1,101 +1,86 @@
-import { useRegister } from '../hooks/api/useRegister';
-import { RegisterFormFields, ResponseDataDto } from '../dtos/register-user.dto';
+import { FormControl, FormLabel, FormErrorMessage, Input, Button, VStack } from '@chakra-ui/react';
+import { useFormik } from 'formik';
 import { AxiosError } from 'axios';
+import { ApplicationErrorDto } from '@shared/dtos/application-error';
+import { reverse } from '@shared/helpers/routing/reverse';
+import { mapValidationErrors } from '@shared/mappers/application-error';
+import { applicationErrorHandler } from '@shared/helpers/application-error';
+import { useRegister } from '@users/hooks/api/useRegister';
+import { registerValidation } from '@users/validation';
+import { RegisterUserDto } from '@users/dtos';
 
-const initialValues: RegisterFormFields = {
+interface RegisterFormValues {
+  email: string;
+  password: string;
+  repeatPassword: string;
+}
+
+const initialValues: RegisterFormValues = {
   email: '',
   password: '',
   repeatPassword: '',
 };
 
-const ValidationSchema = object().shape({
-  email: string().required('Required').email('Invalid email'),
-  password: string().required('Required'),
-  repeatPassword: string()
-    .required('Required')
-    .oneOf([ref('password')], 'Passwords do not match'),
-});
-
 export const RegisterForm: React.FC = () => {
   const navigate = useNavigate();
-  const { mutate } = useRegister();
+  const { mutate: register } = useRegister();
 
   const navigateToLogin = (): void => navigate(reverse('users.login'));
 
-  const handleSubmitForm = (values: RegisterFormFields): void => {
-    const registerUser = { email: values.email, password: values.password };
-    mutate(registerUser, { onError });
-  };
-
-  const formik: FormikProps<RegisterFormFields> = useFormik<RegisterFormFields>({
-    initialValues,
-    onSubmit: handleSubmitForm,
-    validationSchema: ValidationSchema,
-  });
+  const handleSubmitForm = (values: RegisterFormValues): void =>
+    register({ email: values.email, password: values.password }, { onError: handleError });
 
-  const onError = (error: AxiosError<ResponseDataDto, unknown>): void => {
-    const applicationErrorHandler = new ApplicationErrorHandler(error?.response?.data.code);
-    applicationErrorHandler.onGenericEmailUnavailable(() => {
-      formik.setFieldError(
-        'email',
-        applicationErrors[ApplicationErrorCode.GENERIC_EMAIL_UNAVAILABLE]
-      );
+  const { errors, touched, values, handleChange, handleSubmit, setErrors, setFieldError } =
+    useFormik<RegisterFormValues>({
+      initialValues,
+      onSubmit: handleSubmitForm,
+      validationSchema: registerValidation,
     });
-  };
+
+  const handleError = (error: AxiosError<ApplicationErrorDto<any, any>, unknown>): void =>
+    applicationErrorHandler<RegisterUserDto>()
+      .onGenericValidationFailed(({ details }) => setErrors(mapValidationErrors(details)))
+      .onGenericEmailUnavailable(({ message }) => setFieldError('email', message))
```

Comment:
Jednak uzywajmy tych wiadomosci z backendu. Bo zeby ich nie uzywac i jednoczesnie nie miec smietnika w kodzie trzebaby wprowadzic kolejny koncept, mysle ze to bedzie za duzo